### DISCUSSION TITLE
Is it possible to expose the shared data URL as JSON?

### DISCUSSION THREAD

**Discussion by angelhodar** [umami-software/umami#641] [Q&A]
Initially I thought if it was possible to expose an API of the DB to, for example, get the views in a page (like some blogs that shows the view count). Something like [this](https://plausible.io/docs/stats-api#:~:text=The%20Plausible%20Stats%20API%20offers,your%20sites%20through%20the%20API.) but with less complexity and features to start.

But as we have the possibility to share analytics by URL, what about exposing the URL data as JSON to be consumed like a single endpoint? Or maybe just create another colum with an autogenerated api key for a website, which would be like the "token" to be used as authorization by my website to show the view count for each blog post.

**Comment by mikecao** [comment]
You wouldn't specifically need to go through a shared url. If you can log in as the admin or website owner, you can pull stats from the api.

**Comment by joshinat0r** [comment]
I was looking for something similar a while ago for an 11ty page (an SSG).
Wrote myself a small [JS helper](https://gist.github.com/joshinat0r/8f3428ab0a1de834d07efd29703c2c3f) for that, essentially talks with the database like the backend would and spits out JSON.
Just don't use it in the frontend, with 11ty I'm fetching those stats during the build.
